### PR TITLE
Implemented: functionality to allow user to add product on scanning (#324)

### PR DESCRIPTION
**Pull request by amansinghbais**
### Related Issues

Related Issue #324

### Short Description and Why It's Useful

- Added save button in toast to allow user to add product if not found in shipment or POs.
- Added check to not scan if no value provided in sku searching.

### Screenshots of Visual Changes before/after (If There Are Any)

Before
Screenshot from 2024-03-22 12-08-11

After
Screenshot from 2024-03-22 12-06-54
Screenshot from 2024-03-22 14-30-54

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/receiving#contribution-guideline)

### CODE REVIEW

File: src/store/modules/order/actions.ts [hunk 1]
```diff
@@ -46,10 +46,10 @@ const actions: ActionTree<OrderState, RootState> = {
     if (item) {
       item.quantityAccepted = item.quantityAccepted ? parseInt(item.quantityAccepted) + 1 : 1;
       commit(types.ORDER_CURRENT_UPDATED, state.current )
-      showToast(translate("Scanned successfully.", { itemName: payload }))
-    } else {
-      showToast(translate("Scanned item is not present within the shipment:", { itemName: payload }))
+      return { isUpdated: true, itemName: payload }
```
Comment (ymaheshwari1) [inline src/store/modules/order/actions.ts:49]: No need to return `itemName` from here, as we already have `itemName` in the calling method.

File: src/store/modules/order/actions.ts [hunk 2]
```diff
@@ -46,10 +46,10 @@ const actions: ActionTree<OrderState, RootState> = {
...
     }
+
+    return { isUpdated: false, itemName: payload }
```
Comment (ymaheshwari1) [inline src/store/modules/order/actions.ts:52]:
```suggestion
    return { isProductFound: false, itemName: payload }
```

File: src/views/PurchaseOrderDetail.vue [hunk 3]
```diff
@@ -190,6 +190,7 @@ import LocationPopover from '@/components/LocationPopover.vue'
 import ImageModal from '@/components/ImageModal.vue';
 import { copyToClipboard, hasError, productHelpers } from '@/utils';
 import { Actions, hasPermission } from '@/authorization'
+import { showToast } from '@/utils';
```
Comment (ymaheshwari1) [inline src/views/PurchaseOrderDetail.vue:193]: Use a single import for utils

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/order/actions.ts (hunk 2 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: @amansinghbais

Comment (amansinghbais) [reply src/store/modules/order/actions.ts]: Updated sir. You can verify.

**Review by ymaheshwari1 — APPROVED** [review]

**Pull request merged**